src/urllib3/request.py

```diff
@@ -84,6 +86,21 @@ def request(
 
         urlopen_kw["request_url"] = url
 
+        if json is not None and body is not None:
+            raise TypeError(
+                "request got values for both 'json' and 'body', can only specify one."
+            )
+
+        if json is not None:
+            if headers is None:
+                headers = self.headers.copy()
+            if not ("content-type" in map(str.lower, headers.keys())):
+                headers["Content-Type"] = "application/json"
+
+            body = json_serializer.dumps(
+                json, separators=(",", ":"), ensure_ascii=False
+            ).encode("utf-8")
```

Comment:
Unsure if we need to configure anything special in the encode() case like `surrogatepass`? What does Requests do?

Reply:
Nope this is their logic, they are not using `errors="surrogatepass"`
```python
        if not data and json is not None:
            # urllib3 requires a bytes-like body. Python 2's json.dumps
            # provides this natively, but Python 3 gives a Unicode string.
            content_type = 'application/json'

            try:
              body = complexjson.dumps(json, allow_nan=False)
            except ValueError as ve:
              raise InvalidJSONError(ve, request=self)

            if not isinstance(body, bytes):
                body = body.encode('utf-8')
```